src/vs/editor/contrib/hover/browser/markdownHoverParticipant.ts

```diff
@@ -44,23 +55,56 @@ export class MarkdownHover implements IHoverPart {
 	}
 }
 
+export class VerboseMarkdownHover extends MarkdownHover {
+
+	constructor(
+		owner: IEditorHoverParticipant<MarkdownHover>,
+		range: Range,
+		contents: IMarkdownString[],
+		isBeforeContent: boolean,
+		ordinal: number,
+		public readonly provider: HoverProvider | undefined,
+		public readonly verbosityMetadata: HoverVerbosityMetadata | undefined,
+	) {
+		super(owner, range, contents, isBeforeContent, ordinal);
+	}
+}
+
+interface MarkdownFocusMetadata {
+	index: number | undefined;
+	element: HTMLElement | undefined;
+	focusRemains: boolean;
+}
+
 export class MarkdownHoverParticipant implements IEditorHoverParticipant<MarkdownHover> {
 
 	public readonly hoverOrdinal: number = 3;
+	private _anchor: HoverAnchor | undefined;
+	private _context: IEditorHoverRenderContext | undefined;
+
+	private _providers: (HoverProvider | undefined)[] = [];
+	private _focusMetadata: MarkdownFocusMetadata = {
```

Comment:
Why is this "metadata"? Isn't `providers` or `anchor` also metadata then? I'd call this `FocusedHoverInfo`